R/zchunk_L251.en_ssp_nonco2.R

```diff
@@ -35,106 +40,232 @@ module_emissions_L251.en_ssp_nonco2_DISABLED <- function(command, ...) {
-    L161.SSP2_EF <- get_data(all_data, "temp-data-inject/L161.SSP2_EF")
-    L161.SSP15_EF <- get_data(all_data, "temp-data-inject/L161.SSP15_EF")
-    L161.SSP34_EF <- get_data(all_data, "temp-data-inject/L161.SSP34_EF")
-    L201.nonghg_steepness <- get_data(all_data, "temp-data-inject/L201.nonghg_steepness")
+    get_data(all_data, "common/GCAM_region_names") ->
+      GCAM_region_names
+    get_data(all_data, "emissions/A_regions") ->
+      A_regions
+    get_data(all_data, "temp-data-inject/L161.SSP2_EF") %>%
+      gather(year, value, -GCAM_region_ID, -Non.CO2, -supplysector, -subsector, -stub.technology, -agg_sector) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L161.SSP2_EF
+    get_data(all_data, "temp-data-inject/L161.SSP15_EF") %>%
+      gather(year, value, -GCAM_region_ID, -Non.CO2, -supplysector, -subsector, -stub.technology, -agg_sector) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L161.SSP15_EF
+    get_data(all_data, "temp-data-inject/L161.SSP34_EF")  %>%
+      gather(year, value, -GCAM_region_ID, -Non.CO2, -supplysector, -subsector, -stub.technology, -agg_sector) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L161.SSP34_EF
+    get_data(all_data, "temp-data-inject/L201.nonghg_steepness") ->
+      L201.nonghg_steepness
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+# This section takes in the non-CO2 emissions factors for SSP 1/5, 2, and 3/4 across sectors and
+# interpolates across years 2010-2100 in 5 year segments.
+
+    L161.SSP15_EF %>%
+      # Input data only contains 3 future years.  Need to include years between 2010-2100 in 5 year segments,
+      # and interpolate between each time segment.
+      tidyr::complete(year = SSP_FUTURE_YEARS,
```

Comment:
I usually include it unless it's something I'm using all the time.  It never hurts to let someone reading the code know where a function comes from.